src/getCozyToGoogleStrategy.js

```diff
@@ -0,0 +1,41 @@
+const { ADD_COZY_METADATA } = require('./constants')
+const transpiler = require('./transpiler')
+
+const getCozyToGoogleStrategy = (cozyClient, googleUtils) => ({
+  findRemoteDocument: (cozyContact, googleContacts) => {
+    const resourceName =
+      (cozyContact.cozyMetadata && cozyContact.cozyMetadata.sync.id) || null
+    if (resourceName) {
+      return googleContacts.find(
+        googleContact => googleContact.resourceName === resourceName
+      )
+    }
+
+    return undefined
+  },
+  shouldSave: googleContact => !googleContact,
+  save: cozyContact =>
+    googleUtils.createContact(transpiler.toGoogle(cozyContact)),
+  afterSave: async (cozyContact, googleContact) => {
+    let additionalData = {}
+    if (ADD_COZY_METADATA) {
+      additionalData = {
+        cozyMetadata: {
+          sync: {
+            konnector: 'google',
+            last_sync: Date.now(),
+            etag: googleContact.etag,
+            id: googleContact.resourceName
+          }
+        }
+      }
+    }
+
+    return await cozyClient.save({
```

Comment:
If the function returns a `await` variable, it does not have to be marked `async`. In others words:

```js
function getCozyMetaData({ etag, resourceName }) {
  return {
    sync: {
      konnector: 'google',
      last_sync: Date.now(),
      etag,
      id: resourceName
    }
  };
}

function afterSave (cozyContact, googleContact) {
  if (ADD_COZY_METADATA) {
    return cozyClient.save({
      ...cozyContact, 
      ...getCozyMetadata(googleContact)
    });
  }

  return cozyClient.save(cozyContact);
}
```

Returning a Promise is straight forward:

```js
// useless
async function resolve1 () {
  return await Promise.resolve(1);
}

// straight forward
function resolve2 () {
  return Promise.resolve(2);
}
```

See both implementation in [babeljs.io _try out_ editor](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=IYZwngdgxgBAZgV2gFwJYHsIwE4FMToA2AbrgIwwAUAlDAN4BQMOuyC2WwA7sKsjAAVs6ALaoQuAHR4CJXJTLUA3AwC-DBohQYsMoqQBMVWo2Z42HQcLETp-ffIPK1QA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2017&prettier=true&targets=&version=6.26.0&envVersion=), both functions return a `Promise`.

Reply:
Good point!